clang/Type.d

```diff
@@ -127,8 +127,18 @@ struct Type
 
     @property ArrayType array ()
     {
+        assert(isArray);
```

Comment:
I think it's better to put the assert in a constructor in `ArrayType`. There's nothing from stopping anyone to use `ArrayType` directly without going through `array`.